quickcomm/models.py

```diff
@@ -1,3 +1,125 @@
+# TODO what are the constraints on fields being null?
+
+class Author(models.Model):
+    """An author is a person associated with a user account via a one-to-one
+    relationship."""
+
+    # Example code for a one-to-one relationship taken from
+    # https://simpleisbetterthancomplex.com/tutorial/2016/07/22/how-to-extend-django-user-model.html#onetoone
+    # Author: Vitor Freitas
+    # Site: Simple is Better Than Complex
+    # Taken on: Feb 6, 2023
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    host = models.URLField(validators=[URLValidator])
+    display_name = models.CharField(max_length=100)
+    github = models.URLField(blank=True, null=True, validators=[URLValidator])
+    profile_image = models.URLField(blank=True, null=True, validators=[URLValidator])
+
+    # TODO determine if admins are authors
+    is_admin = models.BooleanField(default=False)
+
+    def follows(self, author):
+        """Returns true if this author (self) follows the given author."""
+        return Follow.objects.filter(follower=self, following=author).exists()
+
+    def following(self, author):
+        """Returns true if this author (self) is followed by the given author."""
+        return Follow.objects.filter(follower=author, following=self).exists()
+
+
+    def __str__(self):
+        return f"{self.display_name} ({self.user.username})"
+
+class Follow(models.Model):
+    """A follow is is a many-to-many relationship between authors representing a
+    follow."""
+
+    follower = models.ForeignKey(Author, on_delete=models.CASCADE, related_name='follower')
+    following = models.ForeignKey(Author, on_delete=models.CASCADE, related_name='following')
+
+    def is_bidirectional(self):
+        """Returns true if the follow is bidirectional."""
+        return self.following.follows(self.follower)
+
+    def __str__(self):
+        return f"{self.follower.__str__()} follows {self.following.__str__()}"
+
+class Post(models.Model):
+    """A post is a post made by an author."""
+
+    class PostType(models.TextChoices):
+        TEXT = 'text/plain'
+        MD = 'text/markdown'
+        PNG = 'image/png;base64'
+        JPG = 'image/jpeg;base64'
+        APP = 'application/base64'
+
+    class PostVisibility(models.TextChoices):
+        PUBLIC = 'PUBLIC'
+        FRIENDS = 'FRIENDS'
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    title = models.CharField(max_length=100)
+    source = models.URLField(blank=True, null=True, validators=[URLValidator])
+    origin = models.URLField(blank=True, null=True, validators=[URLValidator])
+    description = models.CharField(max_length=1000)
+    content_type = models.CharField(max_length=50, choices=PostType.choices)
+    content = models.CharField(max_length=10000)
+    # FIXME categories has to be a list of strings of some sort
```

Comment:
should we create an issue to update our model and fix the category type?

Reply:
This will come down to a design decision:

SQLite does not support arrays natively, so the best options are to:
1. Use PostgreSQL, which supports it
2. Serialize a JSON array into a string

Any thoughts?

Reply:
I like option 2 most since we can just stick to sqlite, and it won't be hard to store/update the information. Does anyone feel differently?